src/main/java/net/masterthought/cucumber/json/deserializers/EmbeddingDeserializer.java

```diff
@@ -25,8 +25,12 @@
     public Embedding deserialize(JsonNode rootNode, Configuration configuration) {
         String data = rootNode.get("data").asText();
         String mimeType = findMimeType(rootNode);
+        String name = null;
+        if (rootNode.has("name")) {
+            name = rootNode.get("name").asText();
+        }
 
-        Embedding embedding = new Embedding(mimeType, data);
+        Embedding embedding = new Embedding(mimeType, data, name);
```

Comment:
reading motivation https://github.com/cucumber/cucumber-jvm/issues/1692 I don't understand what is the name from end user perspective

Reply:
`name` is some kind of a custom identifier for the embedding chosen by the user. Main idea behind this change is to allow to have a meaningful name for the embedding when generating test reports.